policy/data/cache/cache.go

```diff
@@ -0,0 +1,52 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+)
+
+type QueryCache interface {
+	Read(ctx context.Context, query string, variables map[string]interface{}) ([]byte, error)
+	Write(ctx context.Context, query string, variables map[string]interface{}, res []byte) error
+}
+
+type cacheKey struct {
+	query     string
+	variables string
+}
+
+type SimpleQueryCache struct {
+	cache map[cacheKey][]byte
+}
+
+func NewQueryCache() SimpleQueryCache {
+	return SimpleQueryCache{
+		cache: make(map[cacheKey][]byte),
+	}
+}
+
+func (d SimpleQueryCache) Read(ctx context.Context, query string, variables map[string]interface{}) ([]byte, error) {
+	variablesJSON, err := json.Marshal(variables)
+	if err != nil {
+		return nil, err
+	}
+
+	key := cacheKey{query: query, variables: string(variablesJSON)}
```

Comment:
I'd move this line into a func so there's one single place where the key is constructed and can be called from both the `Read` and `Write` methods.